libs/IMUKit/tests/IMUKit_test.cpp

```diff
@@ -0,0 +1,163 @@
+	// void TearDown() override {}
+
+	stub::EventLoopKit stub_event_loop {};
+
+	mock::LSM6DSOX mock_lsm6dsox;
+
+	CoreAccelerometer accel {mock_lsm6dsox};
+	CoreGyroscope gyro {mock_lsm6dsox};
+
+	IMUKit imukit {stub_event_loop, accel, gyro};
+
+	interface::LSM6DSOX::SensorData data
+	{
+		{0.F, 0.F, 0.F}, { 0.F, 0.F, 0.F }
+	};
+};
+
+TEST_F(IMUKitTest, initialization)
+{
+	ASSERT_NE(&imukit, nullptr);
+}
+
+TEST_F(IMUKitTest, computeAnglesNullAccelerations)
+{
+	data = {{0.F, 0.F, 0.F}};
+
+	EXPECT_CALL(mock_lsm6dsox, getData).WillRepeatedly(ReturnRef(data));
+
+	for (auto i = 0; i < 100; ++i) {
+		imukit.computeAngles();
+	}
+
+	auto [pitch, roll, yaw] = imukit.getAngles();
+
+	EXPECT_EQ(pitch, 0);
+	EXPECT_EQ(roll, 0);
+	EXPECT_EQ(yaw, 180);
+}
+
+TEST_F(IMUKitTest, computeAngles1GAccelerationOnXAxis)
+{
+	data = {{1000.F, 0.F, 0.F}};
+
+	EXPECT_CALL(mock_lsm6dsox, getData).WillRepeatedly(ReturnRef(data));
+
+	for (auto i = 0; i < 100; ++i) {
+		imukit.computeAngles();
+	}
+
+	auto [pitch, roll, yaw] = imukit.getAngles();
+
+	EXPECT_TRUE(pitch < -80 && pitch > -100);
+	EXPECT_EQ(roll, 0);
+	EXPECT_EQ(yaw, 180);
+}
+
+TEST_F(IMUKitTest, computeAnglesMinus1GAccelerationOnXAxis)
+{
+	data = {{-1000.F, 0.F, 0.F}};
+
+	EXPECT_CALL(mock_lsm6dsox, getData).WillRepeatedly(ReturnRef(data));
+
+	for (auto i = 0; i < 100; ++i) {
+		imukit.computeAngles();
+	}
+
+	auto [pitch, roll, yaw] = imukit.getAngles();
+
+	EXPECT_TRUE(pitch > 80 && pitch < 100);
+	EXPECT_EQ(roll, 0);
+	EXPECT_EQ(yaw, 180);
+}
+
+TEST_F(IMUKitTest, computeAngles1GAccelerationOnYAxis)
+{
+	data = {{0.F, 1000.F, 0.F}};
+
+	EXPECT_CALL(mock_lsm6dsox, getData).WillRepeatedly(ReturnRef(data));
+
+	for (auto i = 0; i < 100; ++i) {
+		imukit.computeAngles();
+	}
+
+	auto [pitch, roll, yaw] = imukit.getAngles();
+
+	EXPECT_EQ(pitch, 0);
+	EXPECT_TRUE(roll > 80 && roll < 100);
+	EXPECT_EQ(yaw, 180);
+}
+
+TEST_F(IMUKitTest, computeAnglesMinus1GAccelerationOnYAxis)
+{
+	data = {{0.F, -1000.F, 0.F}};
+
+	EXPECT_CALL(mock_lsm6dsox, getData).WillRepeatedly(ReturnRef(data));
+
+	for (auto i = 0; i < 100; ++i) {
+		imukit.computeAngles();
+	}
+
+	auto [pitch, roll, yaw] = imukit.getAngles();
+
+	EXPECT_EQ(pitch, 0);
+	EXPECT_TRUE(roll < -80 && roll > -100);
+	EXPECT_EQ(yaw, 180);
+}
+
+TEST_F(IMUKitTest, computeAngles1GAccelerationOnZAxis)
+{
+	data = {{0.F, 0.F, 1000.F}};
+
+	EXPECT_CALL(mock_lsm6dsox, getData).WillRepeatedly(ReturnRef(data));
+
+	for (auto i = 0; i < 100; ++i) {
+		imukit.computeAngles();
+	}
+
+	auto [pitch, roll, yaw] = imukit.getAngles();
+
+	EXPECT_EQ(pitch, 0);
+	EXPECT_EQ(roll, 0);
+	EXPECT_EQ(yaw, 180);
+}
+
+TEST_F(IMUKitTest, computeAnglesMinus1GAccelerationOnZAxis)
+{
+	data = {{0.F, 0.F, -1000.F}};
+
+	EXPECT_CALL(mock_lsm6dsox, getData).WillRepeatedly(ReturnRef(data));
+
+	for (auto i = 0; i < 100; ++i) {
+		imukit.computeAngles();
+	}
+
+	auto [pitch, roll, yaw] = imukit.getAngles();
+
+	EXPECT_EQ(pitch, 0);
+	EXPECT_EQ(roll, 0);
+	EXPECT_EQ(yaw, 180);
+}
```

Comment:
Just to be sure, none of pitch/roll/yaw change, is that normal for an acceleration on Z axis?

Reply:
Indeed, with current configuration of IMU without magnetometer, yaw orientation is arbitrarily set at 180 in first measurement. While no angular speed is recorded by IMU, yaw will not change. I will add several tests with angular speed change to assure that yaw measurement isn't frozen. 

For pitch & roll, this is the remarkable point where roll and pitch are respectively equals to 0 or 180/0. Adding an epsilon on acceleration on X or Y axis makes roll fall to 179 for instance.